File: src/main/java/me/krzu/hsbmechanics/HSBMechanics.java
package me.krzu.hsbmechanics;

import me.fixeddev.commandflow.CommandManager;
import me.fixeddev.commandflow.annotated.AnnotatedCommandTreeBuilder;
import me.fixeddev.commandflow.annotated.part.PartInjector;
import me.fixeddev.commandflow.annotated.part.defaults.DefaultsModule;
import me.fixeddev.commandflow.bukkit.BukkitCommandManager;
import me.fixeddev.commandflow.bukkit.factory.BukkitModule;
import me.krzu.hsbmechanics.commands.CraftCommand;
import me.krzu.hsbmechanics.listeners.CraftListeners;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

public final class HSBMechanics extends JavaPlugin {

    @Override
    public void onEnable() {
        PartInjector injector = PartInjector.create();
        injector.install(new DefaultsModule());
        injector.install(new BukkitModule());

        AnnotatedCommandTreeBuilder treeBuilder = AnnotatedCommandTreeBuilder.create(injector);

        CommandManager commandManager = new BukkitCommandManager(getName());
        commandManager.registerCommands(treeBuilder.fromClass(new CraftCommand(this)));

        PluginManager pluginManager = getServer().getPluginManager();
        pluginManager.registerEvents(new CraftListeners(this), this);
    }

    @Override
    public void onDisable() {
    }
}

File: src/main/kotlin/com/amniel/hsbmechanics/HSBMechanics.kt
package com.amniel.hsbmechanics

import com.amniel.hsbmechanics.commands.CraftCommand
import com.amniel.hsbmechanics.listeners.CraftListeners
import me.fixeddev.commandflow.annotated.AnnotatedCommandTreeBuilder
import me.fixeddev.commandflow.annotated.part.PartInjector
import me.fixeddev.commandflow.annotated.part.defaults.DefaultsModule
import me.fixeddev.commandflow.bukkit.BukkitCommandManager
import me.fixeddev.commandflow.bukkit.factory.BukkitModule
import org.bukkit.plugin.java.JavaPlugin

class HSBMechanics : JavaPlugin() {

    override fun onEnable() {
        val injector = PartInjector.create()
        injector.install(DefaultsModule())
        injector.install(BukkitModule())

        val treeBuilder = AnnotatedCommandTreeBuilder.create(injector)
        val commandManager = BukkitCommandManager(name)
        commandManager.registerCommands(treeBuilder.fromClass(CraftCommand(this)))

        val pluginManager = server.pluginManager
        pluginManager.registerEvents(CraftListeners(this), this)
    }

    override fun onDisable() {
    }
}
